nix/docker-monolithic/default.nix

```diff
@@ -0,0 +1,464 @@
+{ forDockerFile ? false
+, environment ? "testnet"
+, logLocal ? false
+}:
+
+let
+  pkgs = import ../. { };
+
+  sources = pkgs.iohkNix.sources // pkgs.commonLib.sources;
+  rawNixpkgs = sources.nixpkgs;
+  helperPkgs = import rawNixpkgs {
+    config = { };
+    overlays = [ ];
+  };
+  patchedNixpkgs = helperPkgs.runCommand "nixpkgs-patched" {
+    patches = [ ./nixpkgs.patch ];
+  } ''
+    cp -r ${rawNixpkgs} $out
+    chmod -R +w $out
+    cd $out
+    patchPhase
+  '';
+  overlay = self: super: {
+    deroot = self.runCommandCC "deroot" { } ''
+      mkdir -pv $out/bin/
+      g++ ${./deroot.cpp} -o $out/bin/deroot
+    '';
+  };
+in with import patchedNixpkgs { overlays = [ overlay ]; };
+
+let
+  self = import ../../. { };
+  cardano-node-src = sources.cardano-node;
+  cardano-db-sync-src = sources.cardano-db-sync;
+
+  inherit (pkgs.commonLib.cardanoLib) environments;
+  targetEnv = environments.${environment};
+
+  customQueries = {
+    cexplorerBlockCount = {
+      query = "SELECT COUNT(*) AS count FROM block";
+      metrics = [{
+        count = {
+          usage = "COUNTER";
+          description = "total blocks in table";
+        };
+      }];
+    };
+  };
+
+  configuration = { config, ... }: {
+    imports = [
+      (cardano-node-src + "/nix/nixos")
+      (cardano-node-src + "/nix/nixos")
+      (cardano-db-sync-src + "/nix/nixos")
+    ] ++ (import ../nixos/module-list.nix);
+
+    services.cardano-node = {
+      extraArgs = [ "+RTS" "-N3" "-RTS" ];
+      inherit environment environments;
+      topology = pkgs.commonLib.cardanoLib.mkEdgeTopology {
+        inherit (targetEnv) edgeHost edgeNodes edgePort;
+      };
+      enable = true;
+      socketPath = "/run/cardano-node/node.socket";
+    };
+
+    services.cardano-db-sync = {
+      enable = true;
+      cluster = environment;
+      socketPath = "/run/cardano-node/node.socket";
+      user = "cexplorer";
+      postgres = {
+        user = "cexplorer";
+        database = "cexplorer";
+      };
+    };
+    services.cardano-explorer-api = {
+      enable = true;
+    };
+    services.cardano-submit-api = {
+      enable = true;
+      network = environment;
+      socketPath = "/run/cardano-node/node.socket";
+    };
+
+    services.postgresql = {
+      enable = true;
+      authentication = lib.mkBefore ''
+        local all postgres trust
+      '';
+        enableTCPIP = false;
+        extraConfig = ''
```

Comment:
This postgres config implies a requirement of >= 8 GB system RAM.  Could consider adding a note for that requirement to the appropriate README.